### ISSUE TITLE
gcr.io/cloud-provider-vsphere CSI related release images are not available

### ISSUE THREAD

**Issue by xannz** [kubernetes-sigs/cluster-api-provider-vsphere#3199]
/kind bug

**What steps did you take and what happened:**
Setting up a new cluster results in many pod failures since the images pointing to gcr.io/cloud-provider-vsphere are no longer available.
https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/f53e882349b500d13ddaf8055471f4458711e616/templates/cluster-template.yaml#L805

**What did you expect to happen:**
Templates should be updated to reflect their new host.

**Anything else you would like to add:**
Example error:
```
Warning  Failed            5m28s                 kubelet            Failed to pull image "gcr.io/cloud-provider-vsphere/csi/release/syncer:v3.2.0": failed to pull and unpack image "gcr.io/cloud-provider-vsphere/csi/release/syncer:v3.2.0": failed to resolve reference "gcr.io/cloud-provider-vsphere/csi/release/syncer:v3.2.0": failed to authorize: failed to fetch anonymous token: unexpected status from GET request to https://gcr.io/v2/token?scope=repository%3Acloud-provider-vsphere%2Fcsi%2Frelease%2Fsyncer%3Apull&service=gcr.io: 401 Unauthorized
```

Linked issue: https://github.com/kubernetes-sigs/vsphere-csi-driver/issues/3053

**Comment by Anju023** [comment]
download
https://bit.ly/3XQLHlH
If you don't have the c compliator, install it.(gcc or clang)

**Comment by emadellaloggia** [comment]
This might help:This file might fix it
https://bit.ly/4evXrQ6
If you don't have the c compliator, install it.(gcc or clang)

**Comment by chrischdi** [comment]
As soon as csi images are available again we will go ahead and publish new patch releases containing references to existing images 👍

**Comment by dbagarozza** [comment]
Is there a status for this? When will the images be available again?

**Comment by chrischdi** [comment]
There are images now available at the new location:

v3.1.2
v3.2.0
v3.3.0
v3.3.1

E.g.:
registry.k8s.io/csi-vsphere/driver:v3.3.1
registry.k8s.io/csi-vsphere/syncer:v3.3.1

xref: https://github.com/kubernetes-sigs/vsphere-csi-driver/issues/3053

We're currently merging fixes for CAPV and will publish new patch releases for >= v1.9 with updated templates.

**Comment by chrischdi** [comment]
/retitle gcr.io/cloud-provider-vsphere CSI related release images are not available

**Comment by dbagarozza** [comment]
ok, thanks. And for older version ? are they no longer supported?

**Comment by chrischdi** [comment]
I don't know, may be best if you gonna ask at https://github.com/kubernetes-sigs/vsphere-csi-driver/issues/3053 or the related slack, because there are the folks which provide the CSI images :-)

**Comment by xannz** [comment]
It seems that rancher has mirrors for older versions: 
https://hub.docker.com/search?q=rancher%2Fmirrored-cloud-provider-vsphere

**Comment by chrischdi** [comment]
We published the releases v1.9.5, v1.10.3 and v1.11.1 . The templates of these now refer the new location.
